src/components/Compose.js

```diff
@@ -83,6 +183,10 @@ function Compose(props) {
     setIdentityKey(identities[0].publicKey);
   }
 
+  if (messages.length && identityKey) {
+    setUsersRecentMessages(identityKey);
```

Comment:
What do you think about taking a way way simpler route here: instead of looking up the history from displayed messages (which has its pros!), we could just track the history since the app start. In other words, every posted message (see `postMessage` in `src/redux/actions.js`) would dispatch a redux event to add a message to the history of the channel. This way we save some execution time here without sacrificing much of usability.

Reply:
I'd probably add a `history` field somewhere [here](https://github.com/peerlinks/peerlinks-desktop/blob/master/src/redux/reducers.js#L337) and add another action at the big switch in `channels` in the same file. The action type could be something like `CHANNEL_PUSH_TO_HISTORY`, and the payload would obviously be the message text itself.